app/bundles/CampaignBundle/Tests/Executioner/Logger/EventLoggerTest.php

```diff
@@ -74,6 +74,15 @@ public function testAllLogsAreReturnedWithFinalPersist()
         $this->assertEquals($logCollection->getValues(), $persistedLogs->getValues());
     }
 
+    public function testHydrateContactRotationsForNewLogs()
```

Comment:
You should check two states of call:

setRotation() - once
leadRepository->getContactRotations() - never

and

setRotation() - once
leadRepository->getContactRotations() - once